### PR TITLE
christmas/sorting The first version of sorting on the devstronomy

### PR DESCRIPTION
**Pull request by Jarouch000**

### CODE REVIEW

File: server/frontend/src/components/Planets.js [hunk 1]
```diff
@@ -7,39 +7,51 @@ import dataLoader from './data-loader-json.js'
...
+      'Ring System?': 'Yes/No',
+      'Global Magnetic Field?': 'Yes/No',
+    }
 
-  state = {
-    planets: [],
-    selectedPlanet: null,
-    satellites: []
+    this.state = {
+      planets: List(),
```
Comment (mkrauskopf) [inline server/frontend/src/components/Planets.js:45]: Due to this change, selection of a planet row stopped to work. We need to apply:

```
diff --git a/server/frontend/src/components/Planets.js b/server/frontend/src/components/Planets.js
index 5466eaa..cbeabc4 100644
--- a/server/frontend/src/components/Planets.js
+++ b/server/frontend/src/components/Planets.js
@@ -54,7 +54,7 @@ class Planets extends Component {
   }

   _rowClassName = ({index}) => {
-    if (this.state.selectedPlanet === this.state.planets[index]) {
+    if (this.state.selectedPlanet === this.state.planets.get(index)) {
       return 'selectedRow';
     }
     if (index % 2 === 0) {
```

File: server/frontend/src/components/Planets.js [hunk 2]
```diff
@@ -95,13 +108,16 @@ class Planets extends Component {
...
+               rowGetter={({ index }) => this.state.planets.get(index)}
                rowClassName={this._rowClassName}
                onRowClick={(props) => this.showMoons(props.rowData)}
+               sort={this._sort}
+               sortBy={sortBy}
+               sortDirection={sortDirection}
                >
-          <Column label='Name' dataKey='name' width={70} className='main-column' />
-          <Column label='Moons' dataKey='numberOfMoons' width={70} />
+          <Column label='Name' dataKey='name' disableSort={false} width={70} className='main-column' />
```
Comment (mkrauskopf) [inline server/frontend/src/components/Planets.js]: Why do we need `disableSort={false}` for these two columns? It seems to be the default. Sorting works for me in the same way without this two lines changed.

**Comment by mkrauskopf** [comment]
Thanks for the pull request. It sorts nicely! 👍 Please see the minor comments above.

File: server/frontend/src/components/Planets.js (hunk 2 above)
Comment (Jarouch000) [reply server/frontend/src/components/Planets.js]: Yes, it works without this argument. I think we can remove it.

File: server/frontend/src/components/Planets.js (hunk 1 above)
Comment (Jarouch000) [reply server/frontend/src/components/Planets.js:45]: Of course, get(index) instead [index].

**Pull request merged**